### PR TITLE
Prepare cinnamon theme variants

### PR DESCRIPTION
**Pull request by hsbasu**
- set cinnamon component option to true
- add colorable assets file
- Update the meson build script to create variants
- Follow the way gnome-shell variants are created
- Not using directories cinnamon, cinnamon-dark, dark and default. Can be cleaned up later.

### CODE REVIEW

**Comment by hsbasu** [comment]
Hi @Jupi007 The PR for cinnamon variants you requested. Please have a look.

**Comment by Initu-Castilhos** [comment]
Captura de tela de 2023-03-16 03-32-59
The light version has a lot of contrast between the separator and the corner bar.
Captura de tela de 2023-03-16 03-33-27
The dark version is ok.
Captura de tela de 2023-03-16 03-41-11
The checked and radio are 24px and the radio is light in the dark version. Take a look at this P.R. done: https://github.com/ubuntu/yaru/pull/3794

File: cinnamon-shell/src/sass/_tweaks.scss [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+//
+// Check Boxes: checkboxes and radios are 24x24px
+//
+.check-box,
+.radiobutton {
+  StBin {
+    width: 24px;
+    height: 24px;
```
Comment (Jupi007) [inline cinnamon-shell/src/sass/_tweaks.scss:31]: Can you revert the control's size to 20px like before?

```suggestion
    width: 20px;
    height: 20px;
```

File: cinnamon-shell/src/assets/colorable-assets/checkbox/radio-unchecked.svg
Comment (Jupi007) [inline cinnamon-shell/src/assets/colorable-assets/checkbox/radio-unchecked.svg:1]: @ItzSwirlz noticed that controls was white in the dark theme.

File: cinnamon-shell/src/assets/colorable-assets/checkbox/checkbox-checked-focused.svg
Comment (Jupi007) [inline cinnamon-shell/src/assets/colorable-assets/checkbox/checkbox-checked-focused.svg:1]: You've used our old checkmark, see #3793

File: cinnamon-shell/meson.build
Comment (Jupi007) [inline cinnamon-shell/meson.build:1]: Can you clean in this PR the unused folders, please?

**Review by Jupi007 — CHANGES_REQUESTED** [review]
Hi @hsbasu and thanks a lot for this PR!
This is much appreciated ❤️

I added some requested change.

**Comment by Jupi007** [comment]
>  The light version has a lot of contrast between the separator and the corner bar.

@ItzSwirlz Is this problem reproducible in `master`, or only in this PR?

File: test-yaru
[the remark this answers is not in the document]
Comment (hsbasu) [reply test-yaru:1]: No. I was getting trouble to test the themes as it asks for sudo password every time. So I researched and created it. Kept it because others can simply run it while testing. I don't see any harm keeping it. So I kept it.

[the remark this answers is not in the document]
Comment (Jupi007) [reply test-yaru:1]: Okay, what you can do is create a local git ignore: https://stackoverflow.com/a/1753078
So you don't need to commit this file to our repo 🙂

File: meson_options.txt [hunk 2]
```diff
@@ -9,7 +9,7 @@ option('gtksourceview', type: 'boolean', value: true, description:'build gtksour
 option('metacity', type: 'boolean', value: true, description:'build metacity component')
 option('sounds', type: 'boolean', value: true, description:'build sounds component')
 option('sessions', type: 'boolean', value: true, description:'build sessions component')
-option('cinnamon-shell', type: 'boolean', value: false, description:'build cinnamon component')
+option('cinnamon-shell', type: 'boolean', value: true, description:'build cinnamon component')
```
[the remark this answers is not in the document]
Comment (hsbasu) [reply meson_options.txt:12]: Done.

File: test-yaru
[the remark this answers is not in the document]
Comment (hsbasu) [reply test-yaru:1]: Removed it.

[the remark this answers is not in the document]
Comment (Jupi007) [reply test-yaru:1]: Thanks 👍

File: cinnamon-shell/src/assets/colorable-assets/checkbox/radio-unchecked.svg
Comment (hsbasu) [reply cinnamon-shell/src/assets/colorable-assets/checkbox/radio-unchecked.svg:1]: Done.

File: cinnamon-shell/src/assets/colorable-assets/checkbox/checkbox-checked-focused.svg
Comment (hsbasu) [reply cinnamon-shell/src/assets/colorable-assets/checkbox/checkbox-checked-focused.svg:1]: Done.

**Comment by Jupi007** [comment]
@hsbasu Thanks a lot to have applied my requested changes 🙂
I don't know why, but it looks like you messed up the commits because some of my previous PRs changes pop up here 🤔

@Initu-Castilhos Can you retest this PR, please?
I need to setup a cinnamon VM, but it could be nice to have a quicker feedback, especially for the separator problem.

**Comment by Initu-Castilhos** [comment]
> > A versão light tem muito contraste entre o separador e a barra de canto.
> 
> @Initu-CastilhosEste problema é reproduzível em `master`, ou apenas neste PR?

Master.

**Comment by Initu-Castilhos** [comment]
Captura de tela de 2023-03-16 12-49-17
Captura de tela de 2023-03-16 12-49-25
It's not coloring.

**Comment by Initu-Castilhos** [comment]
Maybe it's better to change the color of the panel and base it on the titlebar color like in Mint-Y.

**Comment by hsbasu** [comment]
@Initu-Castilhos @Jupi007 @ItzSwirlz Corrected everything. Please have a look.

**Comment by Initu-Castilhos** [comment]
Captura de tela de 2023-03-16 16-12-50
Nothing yet.

**Comment by hsbasu** [comment]
> Captura de tela de 2023-03-16 16-12-50 Nothing yet.

It's strange. I rebased all the scss files from the `default` folder. I was able to generate all the variants and assets properly. Let me try again.

**Comment by hsbasu** [comment]
I get it now. Those colours are coming from the sass files and are not using accent_bg_colour as their base. To modify those colours, the sass files need to be modified. I have cross checked all the colorizable assests are taking accent colour.

**Comment by Initu-Castilhos** [comment]
> @Initu-Castilhos @Jupi007Agora eu entendi. Essas cores vêm dos arquivos sass e não estão usando acent_bg_colour como base. Para modificar essas cores, os arquivos sass precisam ser modificados. Eu verifiquei se todos os recursos colorizáveis ​​estão usando cores de destaque. Eu sugeriria ter um commit separado para eles.

I found this project: https://github.com/dgsasha/dg-yaru

**Comment by hsbasu** [comment]
@Jupi007 A gentle reminder on the PR.

**Review by Jupi007 — APPROVED** [review]
Hi @hsbasu,
sorry for the late reply.

Imo, this can be merged.
But I would like to have @Muqtxdir approval before.

**Review by Muqtxdir — APPROVED** [review]
thanks

**Comment by Jupi007** [comment]
Thanks a lot for this PR @hsbasu 👍
And sorry for the delay.

**Comment by hsbasu** [comment]
It's ok.

**Comment by Jupi007** [comment]
@hsbasu it is already merged 🙂

**Comment by Feichtmeier** [comment]
grafik
 :D

**Comment by hsbasu** [comment]
😁 didn't see that it's already merged.

**Comment by Initu-Castilhos** [comment]
It does not complete the installation.

**Comment by Jupi007** [comment]
@Initu-Castilhos I did a fix PR here: #3921
Can you test it?
